### PR TITLE
chore: change NodeIdentity debug

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Remove secret_key from debug print for `NodeIdentity`. The display is using the same fields that are used in the `fmt::Display` trait

### CODE REVIEW

**Review by SWvheerden — DISMISSED** [review]
LGTM

File: comms/src/peer_manager/node_identity.rs [hunk 1]
```diff
@@ -221,3 +221,14 @@ impl fmt::Display for NodeIdentity {
         Ok(())
     }
 }
+
+impl fmt::Debug for NodeIdentity {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("NodeIdentity")
```
Comment (sdbondi) [inline comms/src/peer_manager/node_identity.rs:227]: nit: Debug usually shows all fields, so I think include all fields here but redact the ones that should never be logged:
e.g. see `base_layer/core/src/transactions/transaction_components/unblinded_output.rs`

`.field("secret_key", &"<secret>")`

It's fine for the identity signature to be logged

**Review by sdbondi — COMMENTED** [review]
Looks good, one nit

File: comms/src/peer_manager/node_identity.rs (hunk 1 above)
Comment (sdbondi) [reply comms/src/peer_manager/node_identity.rs:227]: Debug should show all fields, add
`.field("identity_signature", &*acquire_read_lock!(self.identity_signature))`

**Review by stringhandler — APPROVED** [review]

**Review by sdbondi — APPROVED** [review]

**Pull request merged**